### PR TITLE
Added tuples.md and basics.md Italian translations

### PR DESCRIPTION
**Pull request by IowU**

### CODE REVIEW

**Comment by SethTisue** [comment]
know anyone who could review this?

I asked on https://gitter.im/scala/contributors just now, maybe that'll turn up an Italian speaking volunteer

File: _it/tour/basics.md [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+println(newId) // 1
+val newerId: Int = IdFactory.create()
+println(newerId) // 2
+```
+
+Tratteremo gli oggetti più in dettaglio in [seguito](singleton-objects.html).
+
+## Tratti
+
+I tratti sono dei tipi astratti di dati contententi determinati campi e metodi. Secondo le regole di ereditarietà di Scala, una classe può estendere solo *una* classe, ma può estendere più tratti.
```
Comment (dwijnand) [inline _it/tour/basics.md:275]:
```suggestion
I tratti sono dei tipi astratti di dati contenenti determinati campi e metodi. Secondo le regole di ereditarietà di Scala, una classe può estendere solo *una* classe, ma può estendere più tratti.
```

File: _it/tour/basics.md [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+{% scalafiddle %}
+```tut
+trait Greeter {
+  def greet(name: String): Unit =
+    println("Hello, " + name + "!")
+}
+```
+
+Puoi estendere i tratti cone la parola chiave `extends` e sovrascrivere (override) un'implementazione con la parola chiave `override`: 
+You can extend traits with the `extends` keyword and override an implementation with the `override` keyword:
```
Comment (dwijnand) [inline _it/tour/basics.md:296]:
```suggestion
```

File: _it/tour/basics.md [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+```
+{% endscalafiddle %}
+
+Nell'esempio sopra, `DefaultGreeter` estende un singolo tratto, ma avrebbe potuto estenderne un numero maggiore.
+
+Esamineremo i tratti in dettaglio in [seguito](traits.html).
+
+## Il metodo principale (metodo main)
+
+Il metodo principale è il punto di ingresso di un programma Scala. La Macchina Virtuale Java (Java Virtual Machine) necessita di un particolare metodo princiaple, chiamato `main`, che prende un argomento: un array di stringhe.
```
Comment (dwijnand) [inline _it/tour/basics.md:321]:
```suggestion
Il metodo principale è il punto di ingresso di un programma Scala. La Macchina Virtuale Java (Java Virtual Machine) necessita di un particolare metodo principale, chiamato `main`, che prende un argomento: un array di stringhe.
```

File: _it/tour/tuples.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+```tut
+val ingredient = ("Sugar" , 25)
+```
+
+Questo creerà una tupla contenente un elemento di tipo `String` e uno di tipo `Int`.
+
+Il tipo inferito di `ingredient` è `(String, Int)`, che è un'abbreviazione per `Tuple2[String, Int]`.
+
+Per rappresentare le tuple, scala una serie di classi: `Tuple2`, `Tuple3`, ecc., fino a `Tuple22` (compreso).
```
Comment (dwijnand) [inline _it/tour/tuples.md:32]: "ha"?  
```suggestion
Per rappresentare le tuple, scala ha una serie di classi: `Tuple2`, `Tuple3`, ecc., fino a `Tuple22` (compreso).
```

**Review by dwijnand — CHANGES_REQUESTED** [review]
Looks good to me.  Just spotted a few typos.  Also TIL enunciato/i 😄

File: _it/tour/basics.md [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+
+1. Andare all'indirizzo [https://scalafiddle.io](https://scalafiddle.io).
+2. Incolla `println("Hello, world!")` nella sezione di sinistra.
+3. Clicca __Run__. L'output apparirà nella sezione di destra.
+
+_ScalaFiddle_ è integrato direttamente con alcuni degli esempi di codice di questa documentazione; se vedi il pulsante __RUN__ in uno degli esempi seguenti, cliccalo per testare direttamente il codice. 
+
+## Espressioni
+
+Le espressioni sono degli enunciati calcolabili:
```
[the remark this answers is not in the document]
Comment (IowU) [reply _it/tour/basics.md:31]: What does TIL stand for?

File: _it/tour/basics.md [hunk 6]
```diff
@@ -0,0 +1,334 @@
...
+Oppure nulla:
+
+```tut
+def name: String = System.getProperty("user.name")
+println("Hello, " + name + "!")
+```
+
+Ci sono altre differenze, ma per ora puoi pensare ai metodi come qualcosa di simile alle funzioni.
+
+Anche i metodi possono avere espressioni spalmate su più righe (un blocco di codice):
```
[the remark this answers is not in the document]
Comment (IowU) [reply _it/tour/basics.md:173]: Actually I saw the term in some books, although I thought it was kinda weird too.

File: _it/tour/basics.md [hunk 7]
```diff
@@ -0,0 +1,334 @@
...
+
+```tut
+val greeter = new Greeter("Hello, ", "!")
+greeter.greet("Scala developer") // Hello, Scala developer!
+```
+Tratteremo le classi più in dettaglio in [seguito](classes.html).
+
+## Classi case (case classes)
+
+Scala possiede un tipo speciale di classe chiamata classe "case". Di default, un'istanza di una classe case è immutabile, e sono comparate in base al valore (a differenza delle classi normali, le cui istanze sono comprate in base ai loro riferimenti in memoria). Questo le rende molto utili nel [pattern matching](https://docs.scala-lang.org/tour/pattern-matching.html#matching-on-case-classes).
```
[the remark this answers is not in the document]
Comment (IowU) [reply _it/tour/basics.md:211]: Leggo più spesso "di default" che "per" a dire il vero

File: _it/tour/basics.md (hunk 2 above)
Comment (IowU) [reply _it/tour/basics.md:296]: Meglio lasciare override non tradotto? 
"Puoi estendere i tratti con la parola chiave extends ed effettuare un ovveride di un'implementazione con la parola chiave override"?

File: _it/tour/tuples.md (hunk 4 above)
Comment (IowU) [reply _it/tour/tuples.md:32]: I was thinking "usa una serie ...", and I forgot a word. Your suggestion makes more sense though.

File: _it/tour/basics.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (dwijnand) [reply _it/tour/basics.md:31]: Today I Learned.

**Review by dwijnand — APPROVED** [review]
Grazie!

**Review by ekrich — COMMENTED** [review]
You have "cone" vs "con" in that sentence.

File: _it/tour/basics.md [hunk 8]
```diff
@@ -0,0 +1,333 @@
...
+
+{% scalafiddle %}
+```tut
+trait Greeter {
+  def greet(name: String): Unit =
+    println("Hello, " + name + "!")
+}
+```
+
+Puoi estendere i tratti cone la parola chiave `extends` e sovrascrivere (override) un'implementazione con la parola chiave `override`:
```
Comment (dwijnand) [inline _it/tour/basics.md:295]:
```suggestion
Puoi estendere i tratti con la parola chiave `extends` e sovrascrivere (override) un'implementazione con la parola chiave `override`: 
```

**Comment by IowU** [comment]
> You have "cone" vs "con" in that sentence.

Fixed

**Comment by SethTisue** [comment]
mind rebasing? that'll pull in #1832 and #1834, so then we'll see whether the CI failure here has anything to do with your changes or not

**Comment by IowU** [comment]
> 
> 
> mind rebasing? that'll pull in #1832 and #1834, so then we'll see whether the CI failure here has anything to do with your changes or not

I read the log from the failure and it seems it can't resolve some link to documents that I still have not translated. What does rebase mean?

**Comment by dwijnand** [comment]
No, the failures were for links that were broken already.  Those should be now fixed.  So if you could perform a git rebase of your branch onto the current HEAD of master we could get a clearer picture of warnings.

**Comment by IowU** [comment]
> No, the failures were for links that were broken already. Those should be now fixed. So if you could perform a git rebase of your branch onto the current HEAD of master we could get a clearer picture of warnings.

@dwijnand I should have rebased it, but let me know. I'm new to git and its world

**Comment by dwijnand** [comment]
Looks like you still need to push (with `--force`) the rebased branch.  Then we can have a look if it's all good.

**Comment by IowU** [comment]
Done

**Comment by dwijnand** [comment]
😞  still not seeing anything on GitHub...

**Comment by dwijnand** [comment]
Are you rebasing the branch used for https://github.com/scala/docs.scala-lang/pull/1828 ?  You need to rebase your it-tour branch in order to fix this PR.  If you want in a few hours I can fix up all your PRs, seeing as they're in a chain, which makes it more complicated.. (need to use `git rebase --onto ...`).

**Comment by IowU** [comment]
I have rebased the it-tour branch and pushed it. Not sure what I am doing wrong tbh

**Comment by SethTisue** [comment]
@dwijnand where does this PR stand?

**Comment by dwijnand** [comment]
Folding into #1828.

**Pull request closed without merging**